### PR TITLE
Add IPython 6.2.1 with jupyterhub 0.8.1.

### PR DESCRIPTION
**Pull request by gatagat**
This includes:

- current IPython (v6.2.1) bundled with the notebook as well as nbconvert, nbformat, jupyter and all the dependencies
- current jupyterhub (v0.8.1) with dependencies

Both rely on Python 3.6.4 (included as well).

### CODE REVIEW

File: easybuild/easyconfigs/c/configurable-http-proxy/configurable-http-proxy-3.1.1-foss-2017a-nodejs-8.9.4.eb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ Developed as a part of the Jupyter Hub multi-user server."""
+
+toolchain = {'name': 'foss', 'version': '2017a'}
+
+dependencies = [
+    ('nodejs', nodejsver),
+]
+
+source_urls = ['https://github.com/jupyterhub/configurable-http-proxy/archive/']
+sources = ['%(version)s.tar.gz']
```
Comment (verdurin) [inline easybuild/easyconfigs/c/configurable-http-proxy/configurable-http-proxy-3.1.1-foss-2017a-nodejs-8.9.4.eb]: Please add an `SHA256` checksum with `--inject-checksums`

File: easybuild/easyconfigs/i/IPython/IPython-6.2.1-foss-2017a-Python-3.6.4.eb [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    ('ZeroMQ', '4.2.2'),
+]
+
+exts_defaultclass = 'PythonPackage'
+exts_filter = ("python -c 'import %(ext_name)s'", '')
+
+exts_list = [
+    # required for ipython
+    ('Pygments', '2.2.0', {
+        'source_urls': ['https://pypi.python.org/packages/source/P/Pygments/'],
```
Comment (verdurin) [inline easybuild/easyconfigs/i/IPython/IPython-6.2.1-foss-2017a-Python-3.6.4.eb]: Please add checksums here in the same way.

File: easybuild/easyconfigs/j/jupyterhub/jupyterhub-0.8.1-foss-2017a-Python-3.6.4.eb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    ('configurable-http-proxy', '3.1.1', '-nodejs-8.9.4'),
+]
+
+# this is a bundle of Python packages
+exts_defaultclass = 'PythonPackage'
+exts_filter = ("python -c 'import %(ext_name)s'", '')
+
+exts_list = [
+    ('pamela', '0.3.0', {
+        'source_urls': ['https://pypi.python.org/packages/source/p/pamela/'],
```
Comment (verdurin) [inline easybuild/easyconfigs/j/jupyterhub/jupyterhub-0.8.1-foss-2017a-Python-3.6.4.eb]: Please also add checksums here.

File: easybuild/easyconfigs/n/nodejs/nodejs-8.9.4-foss-2017a.eb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ for easily building fast, scalable network applications. Node.js uses an 
+ event-driven, non-blocking I/O model that makes it lightweight and efficient, 
+ perfect for data-intensive real-time applications that run across distributed devices."""
+
+toolchain = {'name': 'foss', 'version': '2017a'}
+toolchainopts = {'lowopt': True}
+
+sources = ['node-v%(version)s.tar.gz']
+
+source_urls = ['http://nodejs.org/dist/v%(version)s/']
```
Comment (verdurin) [inline easybuild/easyconfigs/n/nodejs/nodejs-8.9.4-foss-2017a.eb]: Please remove this unnecessary blank line and add a checksum.

File: easybuild/easyconfigs/p/Python/Python-3.6.4-foss-2017a.eb [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+homepage = 'http://python.org/'
+description = """Python is a programming language that lets you work more quickly and integrate your systems
+ more effectively."""
+
+toolchain = {'name': 'foss', 'version': '2017a'}
+toolchainopts = {'pic': True}
+
+source_urls = ['http://www.python.org/ftp/%(namelower)s/%(version)s/']
+sources = [SOURCE_TGZ]
```
Comment (verdurin) [inline easybuild/easyconfigs/p/Python/Python-3.6.4-foss-2017a.eb]: Again checksum required for the main source.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 37 (5 easyconfigs in this PR)
dell.lan - Linux fedora 27, Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz, Python 2.7.14
See https://gist.github.com/351e1e683849f657db0beb12b33408a6 for a full test report.

**Comment by gatagat** [comment]
@verdurin: It seems the automated test of yours failed because of errors not related to this PR (gcc 6.3.0 itself failed to compile), could you check that?

**Comment by verdurin** [comment]
@gatagat yes, I've reported that as an issue. Maybe someone else can run a test build in the meantime...

Thanks for adding the checksums.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2527.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/e61f52e7d51b04be73ecb8c2a03dc78c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2037.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/30ec57d649ed1f8e70529d1acc7d80b3 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @gatagat!

**Pull request merged**